strings.go

```diff
@@ -306,3 +315,51 @@ func keyStmt(key []byte) *ast.GenDecl {
 		}},
 	}
 }
+
+func strConstBlacklist(node ast.Node, info *types.Info, blacklist map[types.Object]struct{}) {
+	strType := types.Typ[types.String]
+	untypedStr := types.Typ[types.UntypedString]
+
+	constCheck := func(node ast.Node) bool {
+		ident, ok := node.(*ast.Ident)
+		if !ok {
+			return true
+		}
+
+		obj := info.ObjectOf(ident)
+		if obj.Type() == strType || obj.Type() == untypedStr {
+			blacklist[obj] = struct{}{}
+		}
+		return true
+	}
+
+	switch x := node.(type) {
+	case *ast.CompositeLit:
+		if _, ok := x.Type.(*ast.ArrayType); !ok {
+			break
+		}
+		for _, elt := range x.Elts {
+			if kv, ok := elt.(*ast.KeyValueExpr); ok {
+				ast.Inspect(kv.Key, constCheck)
+			}
+		}
+	case *ast.ArrayType:
+		if x.Len != nil {
+			ast.Inspect(x.Len, constCheck)
+		}
+	case *ast.GenDecl:
+		if x.Tok != token.CONST {
+			break
+		}
+		for _, spec := range x.Specs {
+			spec, ok := spec.(*ast.ValueSpec)
+			if !ok {
```

Comment:
I'm pretty sure ok must be true given Tok==CONST, so you can just do `spec := spec.(*ast.ValueSpec)`

Reply:
Yes